lib/Sema/MiscDiagnostics.cpp

```diff
@@ -6163,6 +6163,40 @@ diagnoseDictionaryLiteralDuplicateKeyEntries(const Expr *E,
   const_cast<Expr *>(E)->walk(Walker);
 }
 
+// Verifies that references to members are valid under the restrictions of the
+// MemberImportVisibility feature.
+static void diagnoseMissingMemberImports(const Expr *E, const DeclContext *DC) {
```

Comment:
If you go `resolveOverload` route I suggested higher you can add a fix in the same spot that can diagnose it without having to walk AST.